root/usr/lib/systemd/system/ejabberd.service

```diff
@@ -3,18 +3,19 @@ Description=XMPP Server
 After=network.target
 
 [Service]
+Type=forking
 User=ejabberd
 Group=ejabberd
-LimitNOFILE=16000
+LimitNOFILE=65536
+Restart=on-failure
 RestartSec=5
-Environment=LOGS_DIR=/var/log/ejabberd 
-Environment=EJABBERD_CONFIG_PATH=/etc/ejabberd/ejabberd.cfg 
-Environment=SPOOL_DIR=/var/lib/ejabberd
-ExecStart=/bin/sh /opt/ejabberd-16.01/bin/ejabberdctl start
-ExecStop=/opt/ejabberd-16.01/bin/ejabberdctl stop
-ExecReload=/opt/ejabberd-16.01/bin/ejabberdctl reload_config
-Type=oneshot
-RemainAfterExit=yes
+Environment=EJABBERD_LOG_PATH=/var/log/ejabberd/ejabberd.log
+Environment=EJABBERD_CONFIG_PATH=/etc/ejabberd/ejabberd.yml
+Environment=HOME=/var/lib/ejabberd
+ExecStart=/bin/sh -c '/opt/ejabberd-18.04/bin/ejabberdctl start && /opt/ejabberd-18.04/bin/ejabberdctl started'
+ExecStop=/bin/sh -c '/opt/ejabberd-18.04/bin/ejabberdctl stop && /opt/ejabberd-18.04/bin/ejabberdctl stopped'
+PrivateDevices=true
+ProtectSystem=full
```

Comment:
PrivateDevices=true
    ProtectSystem=full

Why are they required?

Reply:
no really clue @DavidePrincipi , maybe @gsanchietti gets one ?

Reply:
I copied it, but I don't remember the source.

But you can use this one: https://github.com/processone/ejabberd/blob/master/ejabberd.service.template

Reply:
Having read the systemd docs... They're appealing, but why have I never seen them used by other services?

Reply:
please check commit comment at https://github.com/processone/ejabberd/commit/ed792274e3d9832929d47627a95ccfbf6182f222#diff-ec85702108f929390de02d2e0b8983e9